mirage/yaml/write_observationlist.py

```diff
@@ -33,117 +99,227 @@ def write_yaml(xml_file, pointing_file, yaml_file, ps_cat_sw=None, ps_cat_lw=Non
-                "      GalaxyCatalog: {}\n".format(GalaxyCatalog),
-                "      ExtendedCatalog: {}\n".format(ExtendedCatalog),
-                "      ExtendedScale: {}\n".format(ExtendedScale),
-                "      ExtendedCenter: {}\n".format(ExtendedCenter),
-                "      MovingTargetList: {}\n".format(MovingTargetList),
-                "      MovingTargetSersic: {}\n".format(MovingTargetSersic),
-                "      MovingTargetExtended: {}\n".format(MovingTargetExtended),
-                "      MovingTargetConvolveExtended: {}\n".format(MovingTargetConvolveExtended),
-                "      MovingTargetToTrack: {}\n".format(MovingTargetToTrack),
-                "      BackgroundRate: {}\n\n".format(BackgroundRate_lw)
+    number_of_exposures = len(xml_dict['ObservationID'])
+
+    # if only one catalog is provided, that catalog will be used for all observations
+    if len(catalog_files) == 1:
+        catalog_files = catalog_files * number_of_exposures
+
+
+    # assemble string that will constitute the yaml content
+    text_out = ["# Observation list created by write_observationlist.py\n\n"]
+
+    from astropy.table import Table
+
+    # loop over exposures
+    for instrument in used_instruments:
+        exposure_indices = np.where(np.array(xml_dict['Instrument']) == instrument)[0]
+        # 1/0
+        for exposure_index in exposure_indices:
+            text = [
+                "Observation{}:\n".format(xml_dict['ObservationID'][exposure_index]),
+                "  Name: '{}'\n".format(xml_dict['ObservationName'][exposure_index]),
+                "  Date: {}\n".format(date),
+                "  PAV3: {}\n".format(PAV3),
             ]
+            if instrument in ['NIRCAM', 'WFSC']:
+                # LW and SW stuff
+                pass
+            elif instrument in ['NIRISS', 'FGS']:
+                text += [
+                    "  Filter: {}\n".format(xml_dict['FilterWheel'][exposure_index]),
+                    "  PointSourceCatalog: {}\n".format(catalog_files[exposure_index]),
+                    "  GalaxyCatalog: {}\n".format(GalaxyCatalog),
+                    "  ExtendedCatalog: {}\n".format(ExtendedCatalog),
+                    "  ExtendedScale: {}\n".format(ExtendedScale),
+                    "  ExtendedCenter: {}\n".format(ExtendedCenter),
+                    "  MovingTargetList: {}\n".format(MovingTargetList),
+                    "  MovingTargetSersic: {}\n".format(MovingTargetSersic),
+                    "  MovingTargetExtended: {}\n".format(MovingTargetExtended),
+                    "  MovingTargetConvolveExtended: {}\n".format(MovingTargetConvolveExtended),
+                    "  MovingTargetToTrack: {}\n".format(MovingTargetToTrack),
+                    "  BackgroundRate: {}\n".format(BackgroundRate),
+                ]
+            # multiply by the number of dithers
+            text_out += text * xml_dict['ImageDithers'][exposure_index]
```

Comment:
I'm not sure that you want a separate entry for each dither, do you? Previously it was one entry per observation (or for NIRCam, one entry per filter configuration in each observation).

Reply:
For simulation purposes we need one .yaml file per position and filter.

Reply:
This is for specifying which catalogs go with which observations within the proposal. Previously the code was set up using the assumption that the same source catalogs would be used for all dithers of a particular filter set within a particular observation. Requiring the user to specify a different source catalog for each dither seems to me like overkill.

Reply:
We will only have one catalogue per observation (actually we want one catalogue per filter to allow for the changes in brightness from one filter to another, but that can be fixed after the .yaml files are generated).  The catalogue will cover an area rather larger than the NIRISS field of view and so the dithers will not carry the field beyond the catalogue area.

Reply:
Right. That's the way the observation list was originally designed to work. (Although you should be able to specify different catalogs for each filter and not have to change the yaml files after they are generated.)